File: Java_Programming/Practise_Programs/PP_104.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_104
//
//  Description       : This program demonstrates how an array
//                      is updated inside a function. The changes
//                      made inside the function are reflected in
//                      the original array.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 12/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

class PP_104
{
//////////////////////////////////////////////////////////////////
//
//  Function Name     : Update()
//
//  Description       : It is used to update each element of the
//                      array by incrementing its value by 1.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 12/08/2026
//
//////////////////////////////////////////////////////////////////

    public static void Update(int Arr[])
    {
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt]++;        
        }
    }

    public static void main(String A[])
    {
        int Brr[] = {10,20,30,40,50};
        int iCnt = 0;

        System.out.println("Array elements before function call : ");

        for(iCnt = 0; iCnt < 5; iCnt++)
        {
            System.out.println(Brr[iCnt]);
        }

        Update(Brr);

        System.out.println("Array elements before function call : ");

        for(iCnt = 0; iCnt < 5; iCnt++)
        {
            System.out.println(Brr[iCnt]);
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      Array elements before function call :
//      10
//      20
//      30
//      40
//      50
//
//      Array elements before function call :
//      11
//      21
//      31
//      41
//      51
//
//////////////////////////////////////////////////////////////////

File: C_Programming/Practise_Programs/PP_104.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_104
//
//  Description       : This program demonstrates the difference
//                      between strlen() and sizeof() for a string.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 13/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<string.h>

int main()
{
    char str[] = "Jay Ganesh";
    int iRet = 0;

    iRet = strlen(str);

    printf("Length of string is : %d\n", iRet);
    
    iRet = sizeof(str);

    printf("Size of string is : %d\n", iRet);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      Length of string is : 10
//      Size of string is : 11
//
//////////////////////////////////////////////////////////////////